src/main/config_provider/type.c

```diff
@@ -0,0 +1,98 @@
+#include "types.h"
+#include "config_provider.h"
+#include "conversions.h"
+
+static PyObject *AerospikeConfigProvider_new(PyTypeObject *type, PyObject *args,
+                                             PyObject *kwds)
+{
+    AerospikeConfigProvider *self =
+        (AerospikeConfigProvider *)type->tp_alloc(type, 0);
+    if (self == NULL) {
+        goto error;
+    }
+
+    static char *kwlist[] = {"path", "interval", NULL};
+    const char *path = NULL;
+    // We take in a python object and do our own input validation
+    // because PyArg_ParseTupleAndKeywords() doesn't check for overflow errors when taking in a C unsigned long
+    // i.e when we pass in a larger value than UINT32_MAX, it will be truncated when assigned to an unsigned long var
+    // in Windows
+    PyObject *py_interval = NULL;
+
+    if (PyArg_ParseTupleAndKeywords(args, kwds, "s|O", kwlist, &path,
+                                    &py_interval) == false) {
+        goto error;
+    }
+
+    uint32_t interval;
+    if (py_interval) {
+        interval = convert_pyobject_to_uint32_t(py_interval, "interval");
+        if (PyErr_Occurred()) {
+            goto error;
+        }
+    }
+    else {
+        interval = AS_CONFIG_PROVIDER_INTERVAL_DEFAULT;
+    }
+
+    self->path = strdup(path);
```

Comment:
Is there max length for the path? If so, using strndup with that limit would be good.

Reply:
The path size doesn't have a limit; when the client config is parsed, the C client will create a copy of this string using `strdup()`

Reply:
When the client is instantiated:

1. https://github.com/aerospike/aerospike-client-python/pull/744/files#diff-88648230768a33aad2fd94662d8ba32af7cc5dffbabcdc002c66620be68ff5b1R585
2. https://github.com/aerospike/aerospike-client-c/blob/14fe70355d653ccfff65118025c78ab76dfbc2cf/src/include/aerospike/as_config.h#L1122
3. https://github.com/aerospike/aerospike-client-c/blob/14fe70355d653ccfff65118025c78ab76dfbc2cf/src/main/aerospike/as_config.c#L359